docs/docs/language_concepts/01_functions.md

```diff
@@ -40,6 +40,19 @@ fn foo(x : Field, y : pub Field) -> Field {
 Note that a `return` keyword is unneeded in this case - the last expression in a function's body is
 returned.
 
+## Main function
+
+If you're writing a binary, the `main` function is the starting point of your program. You can pass all types of expressions to it, as long as they have a fixed size at compile time:
+
+```rust
+fn main(x : Field) // this is fine: passing a Field
+fn main(x : (Field, bool)) // this is also fine: passing a (Field, bool) tuple
+fn main(x : Vec<Field>) // can't compile, has variable size
+fn main(x : str<5>) // this is fine, as long as you pass a string of size 5
+```
+
+Keep in mind [tests](../nargo/02_testing.md) don't differentiate between `main` and any other function. In the snippet above, the `Vec` example passes tests, but won't compile or prove.
```

Comment:
The `Vec` example of `main(x: Vec<Field>)` wouldn't run as a `#[test]` since test functions must have no parameters

Reply:
What I mean is that this passes tests, but won't work for proving:

```rust
use dep::std::collections::vec::Vec;

fn main(x : Vec<Field>) {
    assert(x.get(0) == 1);
    assert(x.get(1) == 2);
}

#[test]
fn test_one() {
    let mut x = Vec::new();
    x.push(1);
    x.push(2);
    main(x);
}
```

But I'll refactor it to be a bit clearer